mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -61,25 +62,76 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.github.fge.lambdas.Throwing;
 import com.github.steveash.guavate.Guavate;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Sets;
-import com.google.common.collect.Sets.SetView;
 
 public class StoreMessageIdManager implements MessageIdManager {
+
+    private static class MessageMoves {
+        private final ImmutableSet<MailboxId> previousMailboxIds;
+        private final ImmutableSet<MailboxId> targetMailboxIds;
+
+        MessageMoves(Collection<MailboxId> previousMailboxIds, Collection<MailboxId> targetMailboxIds) {
```

Comment:
Can you give modifiers on this constructor and the methods, such as other embedded classes.

Reply:
:+1: